oauth.go

```diff
@@ -1,9 +1,46 @@
 package fortnox
 
 import (
+	"context"
+	"errors"
+
+	"github.com/go-redis/redis/v9"
 	"golang.org/x/oauth2"
 )
 
+const (
+	rdsTokenKey = "fortnox_token"
+)
+
+var (
+	ErrNoTokenInTokenStorage = errors.New("fortnox: no token in token storage")
+)
+
+type TokenStorage interface {
+	GetToken(ctx context.Context) (string, error)
+	SetToken(ctx context.Context, token []byte) error
+}
+
+type tokenStorage struct {
+	rc *redis.Client
```

Comment:
Could this be abstracted somehow? To avoid including redis as a library dependency.